src/stream/src/executor/infallible_expr.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use risingwave_common::array::{ArrayRef, DataChunk, Row};
+use risingwave_common::types::Datum;
+use risingwave_expr::expr::Expression;
+use risingwave_expr::ExprError;
+use static_assertions::const_assert;
+
+pub trait InfallibleExpression: Expression {
+    fn eval_infallible(&self, input: &DataChunk, on_err: impl Fn(ExprError)) -> ArrayRef {
```

Comment:
```suggestion
    fn eval_infallible(&self, input: &DataChunk, on_err: impl FnOnce(ExprError)) -> ArrayRef {
```

Reply:
`on_err` is called for each failed row. 🤔

Reply:
Oops. My fault.